### ISSUE TITLE
Subdomains

### ISSUE THREAD

**Issue by prodenx** [samjmckenzie/captcha-manager#1]
Hi

Great work!
Are you sure that captcha solutions will be valid? I think host owner can set allowed subdomains, so localapi subdomain will not be acceptable.

**Comment by samjmckenzie** [comment]
Sorry for the late reply. Have you tested any subdomains? If you could give me an example to work with, that'd be great.

**Comment by prodenx** [comment]
Hi

Yes. I added "captcha.supremenewyork.com" as local resource. I writed C# application which returns captcha page for "captcha.supremenewyork.com". I used solution from this page in real supreme page and got fail. I made the same way for other sites and got fail too.

**Comment by samjmckenzie** [comment]
You're supposed to add localapi.supremenewyork.com, not captcha.supremenewyork.com. I should probably change that, as it's a weird word choice anyway.  
Have you run any of the tests from the test directory?

**Comment by prodenx** [comment]
It does not matter how you call subdomain - captcha or localapi. I did not run your test. I made something like your hervster. My application registers in hosts file localapi.supremenewyork.com domain and responses on requests for http://localapi.supremenewyork.com:8081,  I open browser load this url. Page contains captcha challenge (there is no "key does not match domain" error). I solve captcha and use solution on site. Site can check in what url captcha was solved (https://developers.google.com/recaptcha/docs/verify).

Did you make tests on any sites?

**Comment by samjmckenzie** [comment]
I've just tried on the Supremenewyork domain and it works just fine. https://www.youtube.com/watch?v=cb71HkeGnh0&feature=youtu.be
Also, I'm confused. Are you using this package or are you making your own program?

**Comment by prodenx** [comment]
Yes, I am making my program, You can generate solutions, but supreme rejects solutions generated not on www.supremenewyork.com. Supreme check this via user verify request (https://developers.google.com/recaptcha/docs/verify). Response contains hostname:

{
  "success": true|false,
  "challenge_ts": timestamp,  // timestamp of the challenge load (ISO format yyyy-MM-dd'T'HH:mm:ssZZ)
  "hostname": string,         // the hostname of the site where the reCAPTCHA was solved
  "error-codes": [...]        // optional
}

Hostname will contain localapi.supremenewyork.com.

**Comment by samjmckenzie** [comment]
I see what you mean. I'll see what I can do when I start rewriting some of the code. Creating a proxy in Node and redirecting requests via that might fix it.

**Comment by samjmckenzie** [comment]
Quick update: I'm fairly confident I can fix this by making the user install a Chrome addon that changes the host in the requests made to Google. I'll write the addon as soon as I have some time. Sorry for the big delay.

**Comment by prodenx** [comment]
Hi. No problem:)
I did research of recaptcha js code and found that recaptcha send domain not via host. It uses window.location for this purpose.

**Comment by samjmckenzie** [comment]
Are you sure about that? If so, my method won't be effective at all.

**Comment by prodenx** [comment]
I am not sure. But JS code do this. I did not test all cases.

**Comment by samjmckenzie** [comment]
So, I've looked into this issue some more today. I'm really struggling with coming up with a solution. I was using www.packershoes.com as a reference site. First, I tried changing the `window.location` value of the captcha iframe window using history.replaceState, but you can't change the subdomain with that. Then I tried changing `localapi.packershoes.com` to `www.packershoes.com` in my hosts file but then I can't access the website, since the host will be pointing to local IP.  

Any help with this issue would be greatly appreciated.

**Comment by prodenx** [comment]
Yes, I tried your steps too. They are useless. Proxy + original site domain can be used as solution. Proxy looks at request url: if it has something like `sitedomain.com/captchasolver` then proxy redirects to your captcha server, otherwise - to site server.

**Comment by samjmckenzie** [comment]
I remember trying something with a proxy but I had some issues that I
couldn't fix. Have you tried that yet?
